src/auth/src/credentials.rs

```diff
@@ -253,13 +241,56 @@ pub async fn create_access_token_credential() -> Result<Credential> {
     }
 }
 
+#[derive(Debug, PartialEq)]
+enum AdcPath {
+    FromEnv(String),
+    WellKnown(String),
+    None,
+}
+
+#[derive(Debug, PartialEq)]
+enum AdcContents {
+    Contents(String),
+    FallbackToMds,
+}
+
+fn load_adc() -> Result<AdcContents> {
+    match adc_path() {
+        AdcPath::None => Ok(AdcContents::FallbackToMds),
+        AdcPath::FromEnv(path) => {
+            match std::fs::read_to_string(&path) {
+                Ok(contents) => Ok(AdcContents::Contents(contents)),
+                Err(e) if e.kind() == std::io::ErrorKind::NotFound => Err(CredentialError::new(
+                    false,
+                    Box::from(format!(
+                        "Failed to load Application Default Credentials (ADC) from {path}. Check that the `GOOGLE_APPLICATION_CREDENTIALS` environment variable points to a valid file."
+                    )))),
```

Comment:
nit: maybe move this to a function, so it reads easier?

```suggestion
                Err(e) if e.kind() == std::io::ErrorKind::NotFound => Err(path_not_found(path)),
```

Reply:
Done